### PR TITLE
MDEV-30694: Cross building on x86_64 to arch i686 fails

### PR DESCRIPTION
**Pull request by illuusio**
- [x] *The Jira issue number for this PR is: MDEV-30694*

## Description
Currently cross compilation on x86_64 to arch i686 fails with error:

```
ctype-uca1400data.h
/bin/sh: 1: uca-dump: not found
```

Pull request makes sure that uca-dump is treated correctly when cross compiling MariaDB to another architecture

## How can this PR be tested?
Commit can be tested only with cross compiling. For example with x86_64 machine to i686 and seeing if `uca-dump` fails or not.

## Basing the PR against the correct MariaDB version
- [x] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced*

## Backward compatibility
This should not make any backward compatibility issues

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](CODING_STANDARDS.md) file and my PR conforms to this where appropriate.

### CODE REVIEW

**Comment by helmutg** [comment]
I confirm that my contributions on this PR can be used under the New BSD license (3 clause).

File: strings/CMakeLists.txt [hunk 1]
```diff
@@ -41,7 +41,9 @@ ADD_EXECUTABLE(conf_to_src EXCLUDE_FROM_ALL conf_to_src.c)
 SET_TARGET_PROPERTIES(conf_to_src PROPERTIES EXCLUDE_FROM_DEFAULT_BUILD TRUE)
 TARGET_LINK_LIBRARIES(conf_to_src mysys strings)
 
-ADD_EXECUTABLE(uca-dump uca-dump.c)
+IF(NOT CMAKE_CROSSCOMPILING OR DEFINED CMAKE_CROSSCOMPILING_EMULATOR)
+  ADD_EXECUTABLE(uca-dump uca-dump.c)
```
Comment (LinuxJedi) [inline strings/CMakeLists.txt:45]: Quick question: doesn't this completely disable the `uca-dump` from compiling when cross compiling? If so, isn't this needed for generating `ctype-uca1400data.h` which is needed for compiling?

Comment (LinuxJedi) [reply strings/CMakeLists.txt:45]: Never mind, answered my own question when I figured out the EXPORT part of the CMake.

**Review by LinuxJedi — APPROVED** [review]

**Comment by ottok** [comment]
This was backported downstream in https://salsa.debian.org/mariadb-team/mariadb-server/-/merge_requests/38

**Pull request merged**